gandiva/src/main/java/org/apache/arrow/gandiva/evaluator/JniLoader.java

```diff
@@ -69,6 +75,9 @@ private static void loadGandivaLibraryFromJar(final String tmpDir)
     final String libraryToLoad =
         LIBRARY_NAME + "/" + getNormalizedArch() + "/" + System.mapLibraryName(LIBRARY_NAME);
     final File libraryFile = moveFileFromJarToTemp(tmpDir, libraryToLoad, LIBRARY_NAME);
+    NativeLibrary.getInstance(
```

Comment:
Doesn't the NativeLibrary instance need to be disposed somewhere?

Reply:
Why not also leveraging `java.library.path` system prop ?